101-azurefirewall-create/azuredeploy.json

```diff
@@ -0,0 +1,166 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "virtualNetworkName": {
+      "type": "string",
+      "metadata": {
+        "description": "virtual network name"
+      }
+    },
+    "vnetAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16",
+      "metadata": {
+        "description": "virtual network address range"
+      }
+    },
+    "subnetName": {
+      "type": "string",
+      "defaultValue": "subnet1",
+      "metadata": {
+        "description": "Subnet Name"
+      }
+    },
+    "subnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24",
+      "metadata": {
+        "description": "Subnet prefix"
+      }
+    },
+    "azureFirewallSubnetAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.1.0/24",
+      "metadata": {
+        "description": "AzureFirewallSubnet prefix"
+      }
+    },
+    "firewallName": {
+      "type": "string",
+      "defaultValue": "firewall1",
+      "metadata": {
+        "description": "azure firewall name"
+      }
+    },
+    "publicIPName": {
+      "type": "string",
+      "defaultValue": "publicIP1",
+      "metadata": {
+        "description": "Public IP address name"
+      }
+    }  
+  },
+  "variables": {
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks',parameters('virtualNetworkName'))]",
+    "azureFirewallSubnetName": "AzureFirewallSubnet",
+    "azureFirewallSubnetId": "[concat(variables('vnetId'), '/subnets/', variables('azureFirewallSubnetName'))]",
```

Comment:
simplify and just use resourceId():

resourceId('Microsoft.Network/virtualNetworks/subnets',parameters('virtualNetworkName'), variables('azureFirewallSubnetName'))

Reply:
Done